src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.cs

```diff
@@ -0,0 +1,122 @@
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = [IDEDiagnosticIds.CopilotImplementNotImplementedExceptionDiagnosticId];
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var document = context.Document;
+        var cancellationToken = context.CancellationToken;
+
+        if (document.GetRequiredLanguageService<ICopilotOptionsService>() is not { } copilotOptionsService ||
+            await copilotOptionsService.IsGenerateMethodImplementationOptionEnabledAsync().ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        if (document.GetLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+            await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        // Find the throw statement or throw expression node
+        var throwNode = FindThrowNode(context.Diagnostics[0], cancellationToken);
+        if (throwNode is null)
+            return;
+
+        // Preliminary analysis before registering fix
+        var methodDeclaration = throwNode.Ancestors().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+        if (methodDeclaration != null)
+        {
+            if (methodDeclaration.ExpressionBody != null)
+            {
+                // Arrow expression body not supported yet
+                return;
+            }
+
+            if (context.Document.TryGetSemanticModel(out var semanticModel))
+            {
+                var methodSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration, cancellationToken);
+                if (methodSymbol != null)
+                {
+                    var x = DualChangeAction.New(CSharpAnalyzersResources.Implement_with_Copilot,
+                        // for the non preview
+                        (_, c) => GetDocumentUpdater(context, null)(c),
+                        // no-op for the preview
+                        (_, _) => Task.FromResult(context.Document),
+                        nameof(CSharpAnalyzersResources.Implement_with_Copilot));
+
+                    context.RegisterCodeFix(x, context.Diagnostics);
+                }
+            }
+        }
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        foreach (var diagnostic in diagnostics)
+            await FixOneAsync(editor, document, diagnostic, cancellationToken).ConfigureAwait(false);
+    }
+
+    private static async Task FixOneAsync(
+        SyntaxEditor editor, Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+    {
+        // Find the throw statement or throw expression node
+        var throwNode = FindThrowNode(diagnostic, cancellationToken);
+        if (throwNode is null)
```

Comment:
this should not be possible.  analyzer only reports diagnostics when it has a value throw node to report them on.  so we cannot fail to find them when processing the diag.

Reply:
Discussed this offline seems like @DustinCampbell was viewing them on older commit versions.

All that are marked resolved have already been dealt with.